test/test.refresh.ts

```diff
@@ -152,4 +152,22 @@ describe('refresh', () => {
     assert.strictEqual(headers['x-goog-user-project'], 'my-quota-project');
     req.done();
   });
+
+  it('getRequestHeaders should populate x-goog-user-project header if quota_project_id present and token has not expired', async () => {
```

Comment:
I think we should also add a test that makes sure you get the appropriate header populated after the expiry?

(_sorry for the slow review, we'll work on getting this out the door quickly._).